core/src/main/scala/org/apache/spark/storage/BlockManagerMasterEndpoint.scala

```diff
@@ -48,6 +48,10 @@ class BlockManagerMasterEndpoint(
   // Mapping from block manager id to the block manager's information.
   private val blockManagerInfo = new mutable.HashMap[BlockManagerId, BlockManagerInfo]
 
+  // Mapping from external shuffle service block manager id to the block statuses.
+  private val blockStatusByShuffleService =
+    new mutable.HashMap[BlockManagerId, JHashMap[BlockId, BlockStatus]]
```

Comment:
It is chosen because of consistency with the already existing solution for storing cached blocks:

https://github.com/apache/spark/blob/cc7aea020a45adda9a464b3bb9300a6b35ec77ca/core/src/main/scala/org/apache/spark/storage/BlockManagerMasterEndpoint.scala#L544-L546

I would prefer to have a type alias for `JHashMap[BlockId, BlockStatus]` then it would be a bit  more self-explanatory. Would you prefer to have a comment here?